File: packages/Recognizers/ConversationLanguageUnderstanding/dotnet/CluRecognizerBotComponent.cs
using Microsoft.Bot.Builder;
using Microsoft.Bot.Builder.Dialogs.Declarative;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;

namespace Microsoft.Bot.Components.Recognizers
{
    public class CluRecognizerBotComponent : BotComponent
    {
        public override void ConfigureServices(IServiceCollection services, IConfiguration configuration)
        {
            services.AddSingleton<DeclarativeType>((sp) => new DeclarativeType<CluAdaptiveRecognizer>(CluAdaptiveRecognizer.Kind));
        }
    }
}

File: packages/Recognizers/ConversationLanguageUnderstanding/js/src/cluRecognizerBotComponent.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { BotComponent } from 'botbuilder';
import { ComponentDeclarativeTypes } from 'botbuilder-dialogs-declarative';
import {
  Configuration,
  ServiceCollection,
} from 'botbuilder-dialogs-adaptive-runtime-core';
import { CluAdaptiveRecognizer } from './clu/cluAdaptiveRecognizer';

/**
 * CLU Recognizer BotComponent definition.
 */
export class CluRecognizerBotComponent extends BotComponent {
  /**
   * @param services Services collection to register dependency injection.
   * @param _configuration Configuration for the bot component.
   */
  configureServices(
    services: ServiceCollection,
    _configuration: Configuration,
  ): void {
    services.composeFactory<ComponentDeclarativeTypes[]>(
      'declarativeTypes',
      (declarativeTypes) =>
        declarativeTypes.concat({
          getDeclarativeTypes() {
            return [
              {
                kind: CluAdaptiveRecognizer.$kind,
                type: CluAdaptiveRecognizer,
              },
            ];
          },
        }),
    );
  }
}
